nwb_conversion_tools/baserecordingextractorinterface.py

```diff
@@ -48,12 +48,36 @@ def get_metadata_schema(self):
                 type="array",
                 minItems=1,
                 items={"$ref": "#/properties/Ecephys/properties/definitions/ElectrodeGroup"}
-            )
+            ),
+            Electrodes=dict(
+                type="array",
+                minItems=0,
+                items={"$ref": "#/properties/Ecephys/properties/definitions/Electrodes"}
+            ),
         )
         # Schema definition for arrays
         metadata_schema['properties']['Ecephys']['properties']["definitions"] = dict(
             Device=get_schema_from_hdmf_class(Device),
             ElectrodeGroup=get_schema_from_hdmf_class(ElectrodeGroup),
+            Electrodes=dict(
+                type="object",
+                additionalProperties=False,
+                required=["name"],
+                properties=dict(
+                    name=dict(
+                        type="string",
+                        description="name of this electrodes column"
+                    ),
+                    description=dict(
+                        type="string",
+                        description="description of this electrodes column"
+                    ),
+                    data=dict(
+                        type="array",
+                        description="values for each row in this electrodes column"
```

Comment:
```suggestion
                        description="values for each row in this electrodes column"
                        default=[np.nan]*len(self.recording_extractor.get_channel_ids())
```

Would have to add `numpy as np` to imports at the top - but I suspect it would cause an issue to both to have the `data` field as non-required but lacking any default values (based on lines [595-598](https://github.com/catalystneuro/spikeextractors/blob/master/spikeextractors/extractors/nwbextractors/nwbextractors.py#L595-L598)) of `add_electrodes`).

Reply:
Still doesn't solve the greater issue Saksham is raising regarding the mapping of electrode ids that aren't either in strictly incrementing order from 0, or that span multiple interfaces with mixing/matching ids, but that's a bigger issue to think about.